src/pages/Scan.vue

```diff
@@ -92,25 +92,38 @@ export default {
                     // Success function
                     function(response) {
                         // Add error message for invalid and fail codes.
-                        if (
-                            response.data.invalid.length > 0
-                        ) {
-                            this.showFail();
-                            this.errorMessage = "Please enter a valid voucher code.";
-                        } else if (
-                            response.data.fail.length > 0
-                        ) {
+                        var data = response.data;
+
+                        if (data.invalid.length + data.fail.length === 1) {
+                            // single mismatch handler;
+                            if (data.invalid.length > 0) {
+                                this.showFail();
+                                this.errorMessage = "Please enter a valid voucher code.";
+
+                            } else if (data.fail.length > 0) {
+                                this.showFail();
+                                this.errorMessage = "That voucher may have been used already.";
+                            }
+
+                        } else if (data.invalid.length + data.fail.length > 1) {
+                            // rough multifailure manager
                             this.showFail();
-                            this.errorMessage = "That voucher may have been used already.";
+                            this.errorMessage = `[xXx] ${data.success.length} accepted,
```

Comment:
Isn't that what babel is there for - so we can use these helpful new language features? In my mind it's fine because we're refactoring this anyway.